src/app/pages/swap/components/selected-asset-field.tsx

```diff
@@ -1,46 +1,44 @@
 import { Field } from 'formik';
-import { Box, Flex, HStack, styled } from 'leather-styles/jsx';
-
-import { Flag } from '@app/components/layout/flag';
+import { Box, Flex, HStack } from 'leather-styles/jsx';
 
 interface SelectedAssetFieldProps {
   contentLeft: React.JSX.Element;
   contentRight: React.JSX.Element;
-  icon?: string;
   name: string;
+  showError?: boolean;
 }
 export function SelectedAssetField({
   contentLeft,
   contentRight,
-  icon,
   name,
+  showError,
 }: SelectedAssetFieldProps) {
   return (
     <Flex
       alignItems="center"
       border="1px solid"
-      borderColor="accent.border-default !important"
+      borderColor={showError ? 'error !important' : 'accent.border-default !important'}
       borderRadius="10px"
```

Comment:
As far as the user is concerned, this is the input, because that's how we've styled it. `<input />` elements have many restrictions to their use, so it's common to put an input inside of an another element, but it should still work like an input.

If we make this a `label` with a `htmlFor` attribute that's taken as a prop, that corresponds to the `id` attribute of the input within it, we get the expected behaviour. IMO these are important details, as currently there's an line invisible to the user, where the input will/won't be focused.

https://github.com/leather-wallet/extension/assets/1618764/42611aab-a347-4d12-94c4-18180ec72734

Reply:
Sure, I can change this. As a user myself, I don't think I expect the entire component to trigger the input, but happy to change it.